tests/test_search_requests.py

```diff
@@ -151,30 +174,50 @@ def test_vespa_error_details():
 
 
 def test_filter_profiles_return_different_queries():
-    exact_yql = build_yql(
-        request=SearchParameters(
+    exact_yql = YQLBuilder(
+        params=SearchParameters(
             query_string="test", year_range=(2000, 2023), exact_match=True
         ),
         sensitive=False,
-    )
+    ).to_str()
     assert "stem: false" in exact_yql
     assert "nearestNeighbor" not in exact_yql
 
-    hybrid_yql = build_yql(
-        request=SearchParameters(
+    hybrid_yql = YQLBuilder(
+        params=SearchParameters(
             query_string="test", year_range=(2000, 2023), exact_match=False
         ),
         sensitive=False,
-    )
+    ).to_str()
     assert "nearestNeighbor" in hybrid_yql
 
-    sensitive_yql = build_yql(
-        request=SearchParameters(
+    sensitive_yql = YQLBuilder(
+        params=SearchParameters(
             query_string="test", year_range=(2000, 2023), exact_match=False
         ),
         sensitive=True,
-    )
+    ).to_str()
     assert "nearestNeighbor" not in sensitive_yql
 
     queries = [exact_yql, hybrid_yql, sensitive_yql]
     assert len(queries) == len(set(queries))
+
+
+def test_yql_builder_build_where_clause():
+    params = SearchParameters(query_string="climate")
+    where_clause = YQLBuilder(params).build_where_clause()
+    assert "climate" in where_clause
+
+    params = SearchParameters(
+        query_string="climate", keyword_filters={"family_geography": "SWE"}
+    )
+    where_clause = YQLBuilder(params).build_where_clause()
+    assert "SWE" in where_clause
+
+    params = SearchParameters(query_string="climate", year_range=(2000, None))
+    where_clause = YQLBuilder(params).build_where_clause()
+    assert "2000" in where_clause
```

Comment:
do we want to be more specific with this and the below assertion? this doesn't catch a case where e.g. 2000 ends up in the query string

Reply:
Good shout! I'll add that